### PR TITLE
feat(protocol-designer): add batch edit mix disabled fields

### LINKED ISSUE #7558: Mix batch edit: Disabled fields
# Overview

Like Batch Edit Transfer, Mix has its own logic for disabling/enabling fields based on the contents of the other forms in the multi-selection.

# Acceptance Criteria

- Implement `getMultiSelectDisabledFields` so that for `mix` it implements the logic outlined [in the enabled conditions doc](https://docs.google.com/spreadsheets/d/1dbEeeFj3Bdi9IsRQUl_N-xUr3XnERoJkL2BhEuWO9gw/edit#gid=0)
- If this ticket goes in before #7557, edit that ticket (bc stubbing disabledFields would no longer be necessary). If this ticket goes in after #7557, remove the stub and use the actual results of the new `getMultiSelectDisabledFields`
- Test coverage!

# Specification for field (derived from the spreadsheet)

```
// must be same pipette
aspirate_flowRate
dispense_flowRate

// must be same mix labware
aspirate_delay_checkbox
aspirate_delay_seconds
dispense_delay_checkbox
dispense_delay_seconds
mix_mmFromBottom
mix_touchTip_checkbox
mix_touchTip_mmFromBottom

// never disabled
mix_wellOrder_first
mix_wellOrder_second
blowout_checkbox
blowout_location

// pre-wet tip on spreadsheet, but is not a thing for Mix
// mix fields on spreadsheet, but not a thing for Mix
// air gap on spreadsheet, but not a thing for Mix

// ===========

// these fields aren't in the batch edit Mix form, so they're never disabled (n/a)
times
changeTip
labware
pipette
volume
wells
```

### PR DESCRIPTION
**Pull request by shlokamin**
# Overview
This PR modifies getMultiSelectDisabledFields to return a map of disabled fields for mix.

Closes #7558

# Changelog

Example changelog:
- Add batch edit mix disabled fields

# Review requests
Not actually being used yet, so just review the code + tests and make sure what's there makes sense. 

Also, make sure disabled fields still show up for batch edit transfer forms, as the selector I modified is shared.

It looks like in the mix batch edit form the only time fields should be disabled are when pipettes/labware are different. 

See the [excel sheet](https://docs.google.com/spreadsheets/d/1dbEeeFj3Bdi9IsRQUl_N-xUr3XnERoJkL2BhEuWO9gw/edit#gid=0) for more info.

@IanLondon if you want me to just walk through this with you lmk, it's kinda weird cuz you can't actually verify this works in PD yet.

# Risk assessment
Low, not being used yet.

### CODE REVIEW

File: protocol-designer/src/ui/steps/selectors.js [hunk 1]
```diff
@@ -382,3 +337,85 @@ export const getBatchEditSelectedStepTypes: Selector<
     )
   ).sort()
 })
+
+function getMoveLiquidMultiSelectDisabledFields(forms: Array<FormData>) {
```
Comment (shlokamin) [inline protocol-designer/src/ui/steps/selectors.js:341]: decided not to unit test these helper fns bcuz they're private, we should be testing the result of the selector that consumes these fns, which is what we're doing

File: protocol-designer/src/ui/steps/utils.js [hunk 2]
```diff
@@ -24,7 +24,7 @@ type DisabledFields = {
   [fieldName: string]: string,
 }
 
-const pipetteDifferentDisabledFieldNames: Array<StepFieldName> = [
+const batchEditMoveLiquidPipetteDifferentDisabledFieldNames: Array<StepFieldName> = [
```
Comment (shlokamin) [inline protocol-designer/src/ui/steps/utils.js:27]: renamed these variables to reflect the kind of batch edit form they pertain to

File: protocol-designer/src/ui/steps/utils.js [hunk 3]
```diff
@@ -98,44 +111,62 @@ const fieldsWithDisabledTooltipText = (
     }),
     {}
   )
-export const getPipetteDifferentDisabledFields = (): DisabledFields =>
+
+type BatchEditFormType = 'moveLiquid' | 'mix'
```
Comment (IanLondon) [inline protocol-designer/src/ui/steps/utils.js:117]: I like this type bc maybe we'll have a kind of batch edit form that doesn't translate exactly to a `stepType` some day

**Review by IanLondon — APPROVED** [review]
looks good! I re-traced out the spreadsheet and made sure I understood it correctly bc some stuff is wrong (eg, Air Gap should be N/A not "hide" bc there is no air gap for Mix). The fields you got for same-labware/same-pipette fields are the same as I got!

Refactors of moveLiquid look good too

**Pull request merged**